client-course-schedulizer/src/utilities/hooks/useImportFile.ts

```diff
@@ -60,36 +55,60 @@ export const useImportFile = () => {
       }
       scheduleJSON = csvStringToSchedule(scheduleString);
 
-      // TODO: store in local storage incase prof navigates away while editing.
-      // currently a redundant check
-      if (!isEqual(schedule, scheduleJSON)) {
-        await appDispatch({ payload: { fileUrl: "" }, type: "setFileUrl" });
-        await appDispatch({ payload: { schedule: scheduleJSON }, type: "setScheduleData" });
-      }
-      setIsCSVLoading(false);
+      appDispatch({ payload: { fileUrl: "" }, type: "setFileUrl" });
+      await updateScheduleInContext(
+        schedule,
+        scheduleJSON,
+        appDispatch,
+        setIsCSVLoading,
+        isAdditiveImport,
+      );
     };
   };
 
   return onInputChange;
 };
 
+/**
+ * Update the Schedule information in the context
+ * @param currentSchedule
+ * @param scheduleJSON
+ * @param appDispatch
+ * @param setIsCSVLoading
+ * @param isAdditiveImport
+ *
+ * Ref: https://stackoverflow.com/a/57214316/9931154
+ */
 export const updateScheduleInContext = async (
-  schedule: Schedule,
+  currentSchedule: Schedule,
   scheduleJSON: Schedule,
```

Comment:
I wonder if it might be best to call this parameter `newSchedule` because `scheduleJSON` seems to suggest that `currentSchedule` isn't a JSON, but it is.